samza-core/src/main/java/org/apache/samza/container/RunLoop.java

```diff
@@ -648,8 +648,8 @@ public void run() {
     @Override
     public void onFailure(TaskCallback callback, Throwable t) {
       try {
-        state.doneProcess();
         abort(t);
+        state.doneProcess();
```

Comment:
Could you please put a comment about why `abort` has to go before `doneProcess`?